src/kdtree.rs

```diff
@@ -362,7 +362,7 @@ impl<A: Float + Zero + One, T: std::cmp::PartialEq, U: AsRef<[A]> + std::cmp::Pa
     }
 
     fn belongs_in_left(&self, point: &[A]) -> bool {
-        point[self.split_dimension.unwrap()] < self.split_value.unwrap()
+        point[self.split_dimension.unwrap()] <= self.split_value.unwrap()
```

Comment:
Great find @TYPEmber!
Do you think a test case can be added using the example in #42?

Reply:
I will add some test for this pr.
Further more, I found my last pr isn't suit for negative number case.

I will make a new pr and test later, thanks for your replying.